contracts/UniAgent.sol

```diff
@@ -0,0 +1,112 @@
+
+    constructor(
+        address _owner,
+        address _uniswapPermit2,
+        address _allowanceTarget,
+        IWETH _weth,
+        address payable _feeCollector
+    ) Ownable(_owner) TokenCollector(_uniswapPermit2, _allowanceTarget) {
+        weth = _weth;
+        feeCollector = _feeCollector;
+    }
+
+    receive() external payable {}
+
+    function setFeeCollector(address payable _newFeeCollector) external onlyOwner {
+        if (_newFeeCollector == address(0)) revert ZeroAddress();
+        feeCollector = _newFeeCollector;
+
+        emit SetFeeCollector(_newFeeCollector);
+    }
+
+    function approveTokens(address[] calldata tokens, address[] calldata spenders) external onlyOwner {
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            for (uint256 j = 0; j < spenders.length; ++j) {
+                IERC20(tokens[i]).safeApprove(spenders[j], Constant.MAX_UINT);
+            }
+        }
+    }
+
+    function withdrawTokens(address[] calldata tokens, address recipient) external onlyOwner {
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            uint256 selfBalance = Asset.getBalance(tokens[i], address(this));
+            if (selfBalance > 0) {
+                Asset.transferTo(tokens[i], payable(recipient), selfBalance);
+            }
+        }
+    }
+
+    function swap(
+        RouterType routerType,
+        address inputToken,
+        uint256 inputAmount,
+        bytes calldata payload,
+        bytes calldata userPermit
+    ) external payable override {
+        address routerAddr = _getRouterAddress(routerType);
+
+        if (inputToken.isETH() && msg.value != inputAmount) revert InvalidMsgValue();
+        if (!inputToken.isETH()) {
+            if (msg.value != 0) revert InvalidMsgValue();
+
+            if (routerType == RouterType.UniversalRouter) {
+                // deposit directly into router if it's universal router
+                _collect(inputToken, msg.sender, universalRouter, inputAmount, userPermit);
+            } else {
+                // v2 v3 use transferFrom
+                _collect(inputToken, msg.sender, address(this), inputAmount, userPermit);
+            }
+        }
+        (bool success, bytes memory result) = routerAddr.call{ value: msg.value }(payload);
+        if (!success) {
+            assembly {
+                revert(add(result, 32), mload(result))
+            }
+        }
+
+        emit Swap({ user: msg.sender, router: routerAddr, inputToken: inputToken, inputAmount: inputAmount });
+    }
+
+    function _getRouterAddress(RouterType routerType) private pure returns (address) {
+        if (routerType == RouterType.V2Router) {
+            return v2Router;
+        } else if (routerType == RouterType.V3Router) {
+            return v3Router;
+        } else if (routerType == RouterType.UniversalRouter) {
+            return universalRouter;
+        } else {
+            revert UnknownRouterType();
+        }
```

Comment:
I think solidity would catch the invalid enum value and just the fail the transaction.